onnxruntime/core/providers/cpu/tensor/identity_op.cc

```diff
@@ -17,4 +17,25 @@ ONNX_CPU_OPERATOR_KERNEL(
     KernelDefBuilder().TypeConstraint("T", DataTypeImpl::AllTensorTypes()).Alias(0, 0),
     IdentityOp<false>);
 
+void IdentityOpHelper::CopyInputToOutput(OpKernelContext* context, const Tensor& input, int output_index) {
```

Comment:
this should be moved to some common place? maybe we have already had one, I think, as we should have implementation for copy nodes.

Reply:
ExecutionProvider has CopyTensor but that's used for cross device copying. I could only find the existing implementation from Identity to re-use for this purpose. If there is another copy implementation I can combine them all in framework/utils

Reply:
thanks for confirming that!

Can we move this to a common place then please?

Reply:
What should it be named in that case so there's no confusion with ExecutionProvider::CopyTensor? 

I did want to move it to framework initially, but that made it confusing (to me at least) as to why there were two CopyTensor variants at that level. I left it being associated with the IdentityOp as that provided a clearer context as to what it was doing.

Reply:
We could also rename the CopyTensor that is used for cross-device copying ("CopyTensorAcrossDevice" or something like that). (On a slightly different note: In the long run, we might even benefit from a method that, in cooperation with the allocation-planner, chooses between a real-copy and sharing … a sort of "CopyOrShareTensor(…)", which we can't do yet.)